src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -167,14 +171,48 @@ case class Event(app_id:                   Option[String],
   }
 }
 
+
 object Event {
-  implicit val jsonEncoder: Encoder[Event] = deriveEncoder[Event]
+  case class Lossy(event: Event) extends AnyVal
+  case class Lossless(event: Event) extends AnyVal
 
+  implicit val jsonEncoder: Encoder[Event] = deriveEncoder[Event]
+  implicit val losslessEncoder: Encoder[Lossless] = Encoder.instance { lossless: Lossless => lossless.event.asJson }
+  implicit val lossyEncoder: Encoder[Lossy] = Encoder.instance { lossy: Lossy =>
+    val event = lossy.event
+    var json = event.asJson.asObject.get
```

Comment:
Still `Option.get`!

Reply:
Still `var`! I don't see how this is more idiomatic. What's the point? How much performance does it bring?